tezos/context/src/kv_store/persistent.rs

```diff
@@ -406,16 +520,91 @@ impl Persistent {
         Ok(())
     }
 
+    fn reload_database(&mut self) -> Result<(), IndexInitializationError> {
+        let list_sizes = FileSizes::make_list_from_file(&self.sizes_file);
+
+        let commit_counter = Self::truncate_files_with_correct_sizes(
```

Comment:
@sebastiencs when is this called? on every reload? on every load?

Reply:
I remember that it was changed so that it doesn't happen at startup, but then we need either the tool or a command line flag to enable that check (or even better both).